### ISSUE TITLE
Why do we turn off weight decay for bias and layernorm?

### ISSUE THREAD

**Issue by JoshVarty** [JoshVarty/GoogleQALabeling#13]
This is done while setting up our optimizer:

```Python
            no_decay = ['bias', 'LayerNorm.bias', 'LayerNorm.weight']
            optimizer_grouped_parameters = [
                {'params': [p for n, p in param_optimizer if not any(nd in n for nd in no_decay)], 'weight_decay': 0.8},
                {'params': [p for n, p in param_optimizer if any(nd in n for nd in no_decay)], 'weight_decay': 0.0}
            ]
```

What is the reason for this?

**Comment by JoshVarty** [comment]
Overfitting doesn't arise from the bias apparently: https://stats.stackexchange.com/questions/153605/no-regularisation-term-for-bias-unit-in-neural-network

Not sure why we do it for LayerNorm, though.
